src/engine/api.rs

```diff
@@ -17,6 +17,129 @@ pub fn next_connection_id() -> ConnectionId {
 }
 
 #[derive(Debug, PartialEq, Clone)]
+pub enum ClientMessage {
+    Consumer(ConsumerManagerMessage),
+    Producer(ProducerManagerMessage),
+    Both(ConsumerManagerMessage, ProducerManagerMessage),
+}
+unsafe impl Send for ClientMessage {}
```

Comment:
I see what you're saying. I don't think there was ever a compiler warning about things _not_ being `Send`, I just assumed you had to be explicit about marking things as such before you sent them through a channel. I don't think any of the types would pose a problem, but I'll definitely remove the explicit impls and see what the compiler thinks about it. Specifically, `SocketAddr` is definitely just a dumb value without any fanciness or external references.